docs/libraries/typespec-client-generator-core/reference/linter.md

```diff
@@ -0,0 +1,31 @@
+---
+title: "Linter usage"
+toc_min_heading_level: 2
+toc_max_heading_level: 3
+---
+
+# Linter
+
+## Usage
+
+Add the following in `tspconfig.yaml`:
+
+```yaml
+linter:
+  extends:
+    - "@azure-tools/typespec-client-generator-core/all"
+```
+
+## RuleSets
+
+Available ruleSets:
+
+- `@azure-tools/typespec-client-generator-core/all`
+- `@azure-tools/typespec-client-generator-core/best-practices:csharp`
+
+## Rules
+
+| Name                                                                 | Description                                                             |
+| -------------------------------------------------------------------- | ----------------------------------------------------------------------- |
+| `@azure-tools/typespec-client-generator-core/require-client-suffix`  | Client names should end with 'Client'.                                  |
+| `@azure-tools/typespec-client-generator-core/property-name-conflict` | Avoid naming conflicts between a property and a model of the same name. |
```

Comment:
Is this a required rule or is it a recommendation to avoid property and model having the same name.

Reply:
all rules are in a sense optional its up to the user to enable them(directly or via a ruleset). For azure specs we require that the `data-plane` or `resource-manager` ruleset is used.